jb/child_allowance.py

```diff
@@ -0,0 +1,236 @@
+    "Hispanic status unknown",
+    inplace=True,
+)
+
+# Combine race + hispanic categories
+person["race_hispan"] = "Other or Unknown"
+person.race_hispan.mask(
+    (person.race == "White") & (person.hispan == "Not Hispanic"),
+    "White non-Hispanic",
+    inplace=True,
+)
+person.race_hispan.mask(
+    (person.race == "Black") & (person.hispan == "Not Hispanic"),
+    "Black",
+    inplace=True,
+)
+person.race_hispan.mask(
+    (person.race == "Other") & (person.hispan == "Not Hispanic"),
+    "Other non-Hispanic",
+    inplace=True,
+)
+person.race_hispan.mask(person.hispan == "Hispanic", "Hispanic", inplace=True)
+
+# Relabel sex categories
+person["sex"].mask(person["sex"] == 1, "Male", inplace=True)
+person["sex"].mask(person["sex"] == 2, "Female", inplace=True)
+
+# Create State categories
+person["state"] = (
+    pd.Series(person.statefip)
+    .apply(lambda x: us.states.lookup(str(x).zfill(2)).name)
+    .tolist()
+)
+
+# Define data collected at the SPM unit level
+spmu_cols = [
+    "spmfamunit",
+    "spmwt",
+    "spmftotval",
+    "spmtotres",
+    "spmchxpns",
+    "spmthresh",
+    "year",
+]
+
+spmu = pd.DataFrame(
+    person.groupby(spmu_cols)[
+        ["child_6", "infant", "toddler", "preschool", "person"]
+    ].sum()
+).reset_index()
+
+# Calculate total cost of transfers, and total number of children
+program_cost = mdf.weighted_sum(spmu, "spmchxpns", "spmwt")
+total_child_6 = mdf.weighted_sum(spmu, "child_6", "spmwt")
+childallowance = program_cost / total_child_6
+
+# Create copies of the dataset in which to simulate the policies
+spmu_replace_cost = spmu.copy(deep=True)
+spmu_flat_transfer = spmu.copy(deep=True)
+
+# Generate simulation-level flags to separate datasets,
+# 0 = base case, 1 = cost replacement design, 2 = flat transfer
+spmu["sim_flag"] = "baseline"
+spmu_replace_cost["sim_flag"] = "cc_replacement"
+spmu_flat_transfer["sim_flag"] = "child_allowance"
+
+# Caluclate new income by simulation
+spmu_replace_cost.spmftotval += spmu_replace_cost.spmchxpns
+
+spmu_flat_transfer["childallowance"] = (
+    childallowance * spmu_flat_transfer.child_6
+)
+spmu_flat_transfer.spmftotval += spmu_flat_transfer.childallowance
+
+# Append dataframes
+spmu_sim = pd.concat(
+    [spmu, spmu_replace_cost, spmu_flat_transfer], ignore_index=True
+)
+
+# Create poverty flags on simulated incomes
+spmu_sim["poverty_flag"] = spmu_sim.spmftotval < spmu_sim.spmthresh
+
+# Calculate per person spmftotval (resources)
+spmu_sim["resources_pp"] = spmu_sim.spmftotval / spmu_sim.person
+
+# Construct dataframe to disaggregate poverty flag to person level
+person_sim = person.drop("spmftotval", axis=1).merge(
+    spmu_sim[
+        [
+            "spmfamunit",
+            "year",
+            "poverty_flag",
+            "sim_flag",
+            "spmftotval",
+            "resources_pp",
+        ]
+    ],
+    on=["spmfamunit", "year"],
+)
+
+# Consider sex, race, state heterogeneity
+poverty_rate = (
```

Comment:
Can simplify these statements with a function, e.g.:
```
def pov(groupby, filter=True):
    return person_sim[filter].groupby(groupby).apply(lambda x: mdf.weighted_mean(x, "poverty_flag", "asecwt")).reset_index()
```
(with proper line length)

then you'd have:
```
poverty_rate = pov("sim_flag")
poverty_rate_sex = pov(["sim_flag", "sex"])
...
```
and also, via the optional `filter` arg:
```
poverty_u6 = pov("sim_flag", person_sim.child_6)
```